docs/proposals/federation.md

```diff
@@ -404,36 +486,99 @@ These are often left implicit by customers, but are worth calling out explicitly
-   treatment.  If the federation system fails catastrophically, the
-   underlying Kubernetes clusters should remain independently usable.
+   clusters.  For this reason the federation control plane software
+   should ideally be completely independent of the Kubernetes cluster
+   control software, and look just like any other Kubernetes API
+   client, with no special treatment.  If the federation control plane
+   software fails catastrophically, the underlying Kubernetes clusters
+   should remain independently usable.
 1. Unified monitoring, alerting and auditing across federated Kubernetes clusters.
 1. Unified authentication, authorization and quota management across
    clusters (this is in direct conflict with failure isolation above,
    so there are some tough trade-offs to be made here).
 
-## Proposed High-Level Architecture
-
-TBD: All very hand-wavey still, but some initial thoughts to get the conversation going...
+## Proposed High-Level Architectures
+
+Two distinct potential architectural approaches have emerged from discussions
+thus far:
+
+1. An explicitly decoupled and hierarchical architecture, where the
+Federation Control Plane sits logically above a set of independent
+Kubernetes clusters, each of which is unaware of the other clusters,
+and of the Federation Control Plane itself (other than to the extent
+that it is an API client much like any other).  One possible example
+of this general architecture is illustrated below, and will be
+referred to as the "Decoupled, Hierarchical" approach.
+1. A more monolithic architecture, where a single instance of the
+Kubernetes control plane itself manages a single logicial cluster
+composed of nodes in multiple availablity zones and cloud providers.
+
+A very brief, non-exhaustive list of pro's and con's of the two
+approaches follows.  (In the interestes of full disclosure, the author
+prefers the Decoupled Hierarchical model for the reasons stated below).
+
+1. Failure isolation: The Decoupled Hierarchical approach provides
+    better failure isolation than the Monolithic approach, as each
+    underlying Kubernetes cluster, and the Federation Control Plane,
+    can operate and fail completely independently of each other.
+2. Scalability: Conceptually the Decoupled Hierarchical model wins here, as
+   each underlying Kubernetes cluster can be scaled completely
+   independently w.r.t. scheduling, node state management, monitoring,
+   network connectivity etc. It is even potentially feasible to stack
+   "Ubernetes" federated clusters (i.e. create federations of
+   federations) should scalability of the independent Federation
+   Control Plane become an issue (although the author does not
+   envision this being a problem worth solving in the short term).
+3. Code complexity: I think that an argument can be made both ways
+   here. It depends on whether you prefer to weave the logic for handling
+   nodes in multiple availability zones and cloud providers within
```

Comment:
I suspect multiple cloud providers is a strawman.

If you restrict requirements to "a cluster that spans multiple zones in a region", especially if users just want their stateless web apps to keep running when one AZ goes down (see also: discussion is academic without concrete use cases), then code+operational complexity(1 regional cluster) seems far less than N clusters + federation?